deltacat/compute/compactor_v2/steps/merge.py

```diff
@@ -48,6 +50,22 @@
 logger = logs.configure_deltacat_logger(logging.getLogger(__name__))
 
 
+def _does_hash_bucket_idx_have_compacted_table(input: MergeInput, hb_idx: int) -> bool:
+    return (
+        input.round_completion_info
+        and input.round_completion_info.hb_index_to_entry_range
+        and input.round_completion_info.hb_index_to_entry_range.get(str(hb_idx))
+        is not None
+    )
+
+
+def _get_all_deletes(
```

Comment:
We probably need to consider the DELETE delta size too. If the delete delta is too large, it should affect the remote task memory requirement to make sure it doesn't cause OOM during merge. Maybe we could have a separate PR to after testing on production deltas and address this potential issue.

Reply:
Yes - a separate PR after testing sounds ideal